### PR TITLE
[Slate] Update types to support 0.46.1

### PR DESCRIPTION
**Pull request by benjiro**
To support Slate 0.46.1, if you noticed anything missed I'll correct it.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/ianstormtaylor/slate/blob/master/packages/slate/Changelog.md
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by benjiro** [comment]
Don't merge this yet, I want to allow one of the other owners to review it first.

File: types/slate/slate-tests.ts [hunk 1]
```diff
@@ -362,7 +358,6 @@ editor
     path: 'a',
     offset: 0,
     text: 'text',
-    marks: [Mark.create({type: 'test_mark'})],
```
Comment (sgreav) [inline types/slate/slate-tests.ts]: Why has this test case been removed? I don't see anything in the patch notes indicating this is no longer valid

Comment (benjiro) [reply types/slate/slate-tests.ts]: Looking at the diff of the commit it has been remove from the exposed method.
 
https://github.com/ianstormtaylor/slate/commit/a220cd5ae10c7e5f7df6d018deae0670526dbb73#diff-7ff73d3c1ad52a0d47563bac44a04389

Let me know if im missing something.

Comment (benjiro) [reply types/slate/slate-tests.ts]: @sgreav Have you had a chance to review my command, I would like to get this merged in ASAP. Cheers

**Comment by isubasti** [comment]
@benjiro, my recommendation is to postpone using 0.46.x till it actually works as expected as the current version has major problem with marks ianstormtaylor/slate#2736 (comment)

**Comment by jessetrinity** [comment]
@benjiro @sgreav I am currently waiting for additional owner approval as requested. Let me know if the situation changes. Thanks!

**Comment by benjiro** [comment]
@jessetrinity Yup just sorting out an issue with the library first before we merge this. Hold off for now.

**Comment by RyanCavanaugh** [comment]
@benjiro ping on this one

**Comment by benjiro** [comment]
@RyanCavanaugh 
Going to close this one for now slate 46 was had a few bugs and 47 has already been released. When i get time ill consolidate this PR with 47 changes.

Cheers,
Ben

**Pull request closed without merging**